### PR TITLE
DAOS-7821 pylint: Improve spellings checking and warn on errors.

### PR DESCRIPTION
**Pull request by ashleypittman**
Do not report spelling errors if preceded by a hypen as these are
likely help text or command options rather than actual errors.

When using GitHub actions report failure back into the job.

Skip-build: true
Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

**Review by jolivier23 — APPROVED** [review]

File: src/tests/ftest/cart/util/cart_logparse.py [hunk 1]
```diff
@@ -404,9 +405,10 @@ def free_pointer(self):
         """Return the memory address freed"""
         return self.get_field(-1).rstrip('.')
 
+
 # pylint: disable=too-many-branches
 class StateIter():
-    """Helper class for LogIter to add a statefull iterator.
+    """Helper class for LogIter to add a state-full iterator.
```
Comment (daltonbohning) [inline src/tests/ftest/cart/util/cart_logparse.py:411]: Should this actually be `stateful`? Probably isn't in the dictionary, but I think that's the proper term here

File: src/tests/ftest/cart/util/cart_utils.py [hunk 2]
```diff
@@ -33,9 +33,8 @@ def __init__(self, *args, **kwargs):
         self.provider = None
         self.module = lambda *x: False
         self.supp_file = "/etc/daos/memcheck-cart.supp"
-        self.src_dir = os.path.dirname(os.path.dirname(os.path.dirname(
-                       os.path.dirname(os.path.dirname(os.path.dirname(
-                       os.path.dirname(os.path.abspath(__file__))))))))
+        self.src_dir = os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(
+            os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))))))
```
Comment (daltonbohning) [inline src/tests/ftest/cart/util/cart_utils.py:37]: Not your doing, but this is ugly either way :(

File: src/tests/ftest/cart/util/cart_utils.py [hunk 3]
```diff
@@ -251,10 +250,8 @@ def get_env(self):
 
         # Do not use the standard .log file extension, otherwise it'll get
         # removed (cleaned up for disk space savings) before we can archive it.
-        log_filename = test_name + "_" + \
-                       env_CCSA + "_" + \
-                       env_PHY_ADDR_STR + "_" + \
-                       "output.orterun_log"
+        log_filename = test_name + "_" + env_CCSA + "_" + env_PHY_ADDR_STR + "_" + \
+            "output.orterun_log"
```
Comment (daltonbohning) [inline src/tests/ftest/cart/util/cart_utils.py:254]: If you wanted to be fancy
```suggestion
        log_filename = '_'.join([test_name, env_CCSA, env_PHY_ADDR_STR, "output.orterun_log"])
```

File: utils/cq/daos_pylint.py [hunk 4]
```diff
@@ -228,18 +229,67 @@ def __str__(self):
...
+    # Skip words which appear to be part of a path
+    if f'/{word}/' in code:
+        return True
+    # Skip things are followed by open quotes
+    if f'{word}(' in code:
+        return True
+    # Skip things which look like source files.
+    if f'{word}.c' in code:
+        return True
+    # Skip things are followed by open colon
```
Comment (daltonbohning) [inline utils/cq/daos_pylint.py:278]:
```suggestion
    # Skip things which are followed by open colon
```

**Review by daltonbohning — APPROVED** [review]

**Pull request merged**